### PR TITLE
renames in fet, chisq, ctt, hwe (func/agg), tdt

### PR DESCRIPTION
**Pull request by jbloom22**

### CODE REVIEW

**Comment by jbloom22** [comment]
@tpoterba I'm assigning you since I've incorporated your feedback from when you reviewed these name changes before I separated them from the IR PR. I'll add a discuss post on breaking changes once this goes in.

File: python/hail/expr/aggregators.py [hunk 1]
```diff
@@ -724,9 +724,9 @@ def hardy_weinberg(expr) -> StructExpression:
     Returns
     -------
     :class:`.StructExpression`
-        Struct expression with fields `r_expected_het_freq` and `p_hwe`.
+        Struct expression with fields `r_expected_het_freq` and `p_value_hwe`.
     """
-    t = tstruct(r_expected_het_freq=tfloat64, p_hwe=tfloat64)
+    t = tstruct(r_expected_het_freq=tfloat64, p_value_hwe=tfloat64)
```
Comment (tpoterba) [inline python/hail/expr/aggregators.py]: I don't like this change -- it's not consistent with other statistical test methods.

File: python/hail/expr/functions.py [hunk 2]
```diff
@@ -340,16 +340,16 @@ def bind(f: Callable, *exprs):
 
 
 @typecheck(c1=expr_int32, c2=expr_int32, c3=expr_int32, c4=expr_int32)
-def chisq(c1, c2, c3, c4) -> StructExpression:
+def chi_sq_test(c1, c2, c3, c4) -> StructExpression:
```
Comment (tpoterba) [inline python/hail/expr/functions.py]: should this be chi_squared_test?

File: python/hail/expr/aggregators.py (hunk 1 above)
Comment (jbloom22) [reply python/hail/expr/aggregators.py]: I'd also prefer just `p_value`, but hesitated out of wanting consistency with the field name produced by `variant_qc`, and thinking `p_value` would be too vague in that context. But I suppose there is only one p-value produced by variant qc so it's not ambiguous.

I also don't like the `r` in front of `expected_het_freq`. Where we use `r` in sample_qc, its to denote a ratio of the words that follow (e.g. `r_ti_tv`). But `expected_het_freq` is already the right description.

So I propose to change `p_value_hwe` to `p_value` and `r_expected_het_freq` to ``expected_het_freq`, including in variant_qc. Cool?

File: python/hail/expr/functions.py (hunk 2 above)
Comment (jbloom22) [reply python/hail/expr/functions.py]: We are indeed abbreviating `squared` with `sq`. R does `chisq.test`, `rchisq`, etc. In `scipy.stats` the test is `chisquare` whereas the distribution is `chi2`.

I could be convinced to expand it to `chi_squared_test` but I'm not inclined to expand `chi_sq` to `chi_squared` everywhere it's used as a statistic, so I found this version readable and consistent.

Comment (jbloom22) [reply python/hail/expr/functions.py]: ok, now I'm starting to like `chi_squared_test`, but still want to keep `chi_sq` for the reported statistic. Cool?

File: python/hail/expr/aggregators.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/expr/aggregators.py]: expected het freq _is_ a ratio, no?

Comment (tpoterba) [reply python/hail/expr/aggregators.py]: and I want there to be some hwe identifier for the p value in variantqc

Comment (jbloom22) [reply python/hail/expr/aggregators.py]: yes, but it's NOT the ratio of `expected` to `het_freq` or of `expected_het` to `freq`. It's the ratio of the expected number of hets to the total number of genotypes.

Comment (jbloom22) [reply python/hail/expr/aggregators.py]: In other words, the ratio part is described by `freq`.

Comment (jbloom22) [reply python/hail/expr/aggregators.py]: ok, so I'll make it `p_value_hwe` but only when produced in `variant_qc`. otherwise `p_value`

Comment (tpoterba) [reply python/hail/expr/aggregators.py]: isn't it the ratio of the observed to expected number of hets?

Comment (tpoterba) [reply python/hail/expr/aggregators.py]: and ya that's a good solution re: p value

File: python/hail/expr/functions.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/expr/functions.py]: fine with me

File: python/hail/expr/aggregators.py (hunk 1 above)
Comment (jbloom22) [reply python/hail/expr/aggregators.py]: No. That's the point of my rant here:
https://github.com/hail-is/hail/pull/3825

I propose changing it to `het_freq_hwe` everywhere. It's the expected het frequency under HWE.

Comment (tpoterba) [reply python/hail/expr/aggregators.py]: ahhh. OK!

**Comment by jbloom22** [comment]
that was the plan. done.

**Pull request merged**